doc/10-Channels.md

```diff
@@ -1,88 +1,298 @@
-request and response structure is inspired by JSON-RPC.
+!!! warning
+
+    As this is still an early preview version, things might change.
+    There may still be severe bugs and incompatible changes may happen without any notice.
+
+Channel plugins are independent processes that run continuously, supervised by Icinga Notifications.
+They receive JSON-formatted requests on `stdin` and reply with JSON-formatted responses on `stdout`.
+The request and response structure is inspired by JSON-RPC.
+
+Note that this documentation uses beautified JSON for ease of reading.
+
+For logging or debugging, channels can write to `stderr`, which will be logged by Icinga Notifications.
 
 ### Request
 
-The request must be in JSON format and should contain following keys:
+A channel receives a request as a JSON object with the following fields:
 
 - `method`: The request method to call.
-- `params`: The params for request method.
+- `params`: Optional params for the request method.
 - `id`: Unsigned int value. Required to assign the response to its request as responses can be sent out of order.
 
 Examples:
 
+- Simple request without any `params`:
+  ```json
+  {
+    "method": "Simple",
+    "id" : 1000
+  }
+  ```
+- Request with `params` of different types:
+  ```json
+  {
+    "method": "WithParams",
+    "params": {
+      "foo": 23,
+      "bar": "hello"
+    },
+    "id": 1000
+  }
+  ```
+
+### Response
+
+Each request must be answered by the channel with a response JSON object of the following fields:
+
+- `result`: The result as JSON format. Omitted when the method does not return a value, i.e., for setter calls, or an
+  error has occurred.
+- `error`: The error message. Omitted when no error has occurred.
+- `id`: The request id.
+
+In case of a present `error` value, the `result` field should be empty.
+Successfully responses without a `result` do only contain the `id` field.
+
+Examples:
+
+- Successfully response without a `result` message:
+  ```json
+  {
+    "id": 1000
+  }
+  ```
+- Successfully response with a `result`:
+  ```json
+  {
+    "result": "hello world",
+    "id": 1000
+  }
+  ```
+- Response with an error:
+  ```json
+  {
+    "error": "unknown method: 'Foo'",
+    "id": 1000
+  }
+  ```
+
+### Methods
+
+The following methods must be implemented by a channel.
+
+#### GetInfo
+
+The parameterless `GetInfo` method returns information about the channel.
+
+Its `result` is expected to be a JSON object with the `json` fields defined in the
+[`Info` type](https://pkg.go.dev/github.com/icinga/icinga-notifications/pkg/plugin#Info).
+The `config_attrs` field must be an array of JSON objects according to the
+[`ConfigOption` type](https://pkg.go.dev/github.com/icinga/icinga-notifications/pkg/plugin#ConfigOption).
+Those attributes define configuration options for the channel to be set via the `SetConfig` method.
+Furthermore, they are used to build the channel configuration UI in Icinga Notifications Web.
```

Comment:
I think that `UI` is quite out of place. You might wanted to rephrase it as `... build the UI channel configuration in ...`?

Reply:
I have rephrased the sentence.